src/serde_arrow_type.erl

```diff
@@ -1,19 +1,142 @@
 %%%-----------------------------------------------------------------------------
-%% @doc The `serde_arrow_type' module provides types in order to represent Apache
-%% Arrow specific primitive logical datatypes, as well as Erlang datatypes which
-%% `serde_arrow' supports serializing and deserializing.
+%% @doc Provides representation for as well as functions for working with Arrow
+%% specific datatypes.
+%%
+%% This module provides a longhand (a 2 tuple) as well as a shorthand (an atom)
+%% represenations for representing a primitive types. Additionally it represents
+%% Nested Types (i.e. Lists) as well as Native Types (i.e. Erlang Native Types
+%% that are supported by this package).
+%%
+%% Generally, when we say an array A has type X, we mean that each element in A
+%% has type X. Thus, an array of type of `Int8' would look like `[1, 2, 3]', but
+%% an array of type `List<Int8>' would look like `[[1, 2], [3]]'.
+%%
+%% == Primitive Types ==
+%%
+%% A Primitive Type is any type that refers to a single value (as opposed to a
+%% list of values). Currently it includes Booleans, Signed Integers, Unsigned
+%% Integers, Floats and Binaries. Both a longhand and shorthand syntax had been
+%% provided to initiate a type.
+%%
+%% The following is a comprehensive list of all supported primitive types:
+%%
+%% <ol>
+%%  <li>Booleans</li>
+%%  <li>Signed Integers: Includes `Int 8', `Int 16', `Int 32' and `Int 64'</li>
+%%  <li>
+%%      Unsigned Integers: Includes `UInt 8', `UInt 16', `UInt 32' and `UInt 64'
+%%  </li>
+%%  <li>Floats: Includes `Float 16', `Float 32' and `Float 64'</li>
+%%  <li>Binaries</li>
+%% </ol>
+%%
+%% === Shorthand Syntax ===
+%%
+%% In the shorthand syntax, you can just use a single atom to refer to a type,
+%% such that the atom is `<name><size>'. The name is a single letter that refers
+%% to the type, and the size is that type's size in bits. Take the example of
+%% `Int 8'. Its name is `s', and size is `8'. Thus in shorthand, it is `s8'. In
+%% the case of datatypes without variations, such as Booleans and Binaries, they
+%% are represented by just their datatype. Do note that internally the longhand
+%% syntax is used.
+%%
+%% Here you can find the name of each type:
+%%
+%% <table border="1">
+%%  <tr>
+%%      <td>Type</td>
+%%      <td>Name</td>
+%%      <td>Example Atom</td>
+%%  </tr>
+%%  <tr>
+%%      <td>Booleans</td>
+%%      <td>N/A</td>
+%%      <td>`bool'</td>
+%%  </tr>
+%%  <tr>
+%%      <td>Signed Integer</td>
+%%      <td>`s'</td>
+%%      <td>`s64'</td>
+%%  </tr>
+%%  <tr>
+%%      <td>Unsigned Integer</td>
+%%      <td>`u'</td>
+%%      <td>`u16'</td>
+%%  </tr>
+%%  <tr>
+%%      <td>Float</td>
+%%      <td>`f'</td>
+%%      <td>`f32'</td>
+%%  </tr>
+%%  <tr>
+%%      <td>Binaries</td>
+%%      <td>N/A</td>
+%%      <td>`bin'</td>
+%%  </tr>
+%% </table>
```

Comment:
This table results in invalid output with ExDoc, but has have valid output with EDoc. I will create an new in `rebar3_ex_doc`. How much I would give to write docs in AsciiDoc or even Markdown!

Reply:
I've created an issue for this: https://github.com/starbelly/rebar3_ex_doc/issues/47